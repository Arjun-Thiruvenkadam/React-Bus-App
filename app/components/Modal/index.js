/**
 *
 * Modal
 *
 */

import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Backdrop from '../Backdrop/index';
import './modal.scss';

const Modal = props => (
  <Fragment>
    <Backdrop show={props.show} clicked={props.modalClosed} />
    <div
      className='Modal'
      style={{
        visibility: props.show ? 'visible' : 'hidden',
        opacity: props.show ? '1' : '0',
      }}
    >
      {props.children}
    </div>
  </Fragment>
);

Modal.propTypes = {
  show: PropTypes.bool,
  modalClosed: PropTypes.func.isRequired,
};

export default Modal;
