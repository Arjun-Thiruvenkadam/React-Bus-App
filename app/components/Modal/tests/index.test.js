import React from 'react';
import { render } from 'react-testing-library';
import Modal from '../index';

describe('<Modal />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Modal show modalClosed={null} />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render and match the snapshot with values hidden', () => {
    const {
      container: { firstChild },
    } = render(<Modal show={false} modalClosed={null} />);
    expect(firstChild).toMatchSnapshot();
  });
});
