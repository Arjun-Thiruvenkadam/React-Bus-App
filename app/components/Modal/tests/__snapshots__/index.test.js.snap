// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render and match the snapshot 1`] = `
<div
  class="backdrop"
/>
`;

exports[`<Modal /> should render and match the snapshot with values hidden 1`] = `
<div
  class="Modal"
  style="visibility: hidden; opacity: 0;"
/>
`;
